Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -208,6 +210,31 @@ def check_elisp_root(share, name)
     EOS
   end
 
+  def check_linkage
+    keg = Keg.new(formula.prefix)
+    checker = LinkageChecker.new(keg, formula)
+    s = ""
+
+    if checker.broken_dylibs?
+      s += <<-EOS.undent
+      Broken dylib links found:
+        #{checker.broken_dylibs.to_a * "\n        "}
+      EOS
+    end
+
+    if checker.undeclared_deps?
+      s += <<-EOS.undent
+      Undelcared dependencies found:
+        #{checker.undeclared_deps * "\n        "}
+
+      Formula is required to delcare all direct dependencies.
+      Please check whether these dependencies are intended.
```

Comment:
My suggestion would be something along the lines of: “Please check wether these dependencies are needed or whether some library was automatically (and unintentionally) picked up by the build system of the formula, something that can usually be influenced with `--without-THING` or some equivalent option.” (As a comprehensible explanation might become even longer, it might make sense to provide just a link and to come up with a dedicated page in our documentation.)

Furthermore, `brew linkage --reverse <formula>` will show what binary of the formula is linking to what, so this might help with the diagnosis of this message and could be suggested as well.